### PR TITLE
Add a footer

### LINKED ISSUE #108: Add a footer to each page
# Add a footer to each page

## Describe the feature

We should have a footer on each page with links to GH, discord, docs, transifex, opeNode (per their TOS), and etc

## Reasons to add this feature

so people can find things easierly

we have no links to the docs anywhere rn

plus:
> ***Make sure to add a link to *www\.openode.io* available at the provided repository URL. Example: \<a href="https:/\/www.openode.io/">Cloud hosted on opeNode.io\</a>***
> As we provide free hosting for open source projects, this is a good way to help us monetise our services.

this needs to be added somewhere as well

## Alternatives to this feature

navigation bar? maybe, but most links won't be internal and a nav bar is typically for internal links only

## Additional information, if any

apparently partials inherit the parent scope, so I can use variables like `{{ message }}` in it automatically. i did not think that worked

### PR DESCRIPTION
**Pull request by RedGuy12**
# Add a footer

## Resolves

closes #108 

## Changes

* add todo comment to discord auth
* add missing string to header for a11y
* minor l10n system changes
* add a footer

## Reason for changes

bc of opeNode's tos
bc I have been wanting to for a while
so we can link to places that would otherwise be hard to find

## Tests

yes

### todo

- [x] move css to tailwind @apply
- [x] mobile
- [x] retron says too squished
- [ ] hide legal info if trademarks are not on page. might not end up being in this pr

### CODE REVIEW

**Comment by wgyt** [comment]
> * [ ]  move css to tailwind (at)apply
you just pinged the (at)apply org

File: _locales/en_US.json [hunk 1]
```diff
@@ -371,7 +371,102 @@
...
+	"footerScratchLegal": {
+		"string": "Scratch is developed by the Lifelong Kindergarten Group at the MIT Media Lab."
+	},
+	"footerSeeOther": { "string": "See {0}." },
+	"footerGoogleLegal": {
+		"string": "Google and the Google logo are registered trademarks of Google LLC in some countries."
+	},
+	"footerReplitLegal": {
+		"string": "Replit and the Replit logo are registered trademarks of Replit, Inc. in some countries."
+	},
```
Comment (wgyt) [inline _locales/en_US.json:462]: Is there actual text published by Google and Replit that we could use?

File: src/auth/discord/index.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ dotenv.config();
 const client = {
 	name: "Discord",
 	link: "http://discord.com/api/oauth2/authorize?client_id=871197807883739187&redirect_uri=https%3A%2F%2Fauth.onedot.cf%2Fauth%2Fdiscord&response_type=code&scope=identify%20connections%20email&state={{url}}",
-	icon: "https://discord.com/assets/3437c10597c1526c3dbd98c737c2bcae.svg",
+	icon: "https://discord.com/assets/3437c10597c1526c3dbd98c737c2bcae.svg", // todo: https://discord.com/branding
```
Comment (wgyt) [inline src/auth/discord/index.js:9]: Please add this to the list of TODO's

File: src/partials/footer.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
+<footer>
```
Comment (wgyt) [inline src/partials/footer.html:1]: can you give a preview image?

**Comment by RedGuy12** [comment]
> > * [ ]   move css to tailwind (at)apply
> 
> you just pinged the (at)apply org

oh oops lol

looks like both members are inactive tho which is good

File: _locales/en_US.json (hunk 1 above)
Comment (RedGuy12) [reply _locales/en_US.json:462]: * Google just asks us to mention how it is a "registered trademark of Google LLC"
* Replit just says it is a trademark, never says we can even use it, so I only added it to be safe

File: src/auth/discord/index.js (hunk 2 above)
Comment (RedGuy12) [reply src/auth/discord/index.js:9]: not in the scope of this pr

Comment (wgyt) [reply src/auth/discord/index.js:9]: make a new issue cc @retronbv

File: _locales/en_US.json [hunk 4]
```diff
@@ -371,7 +371,102 @@
...
+	"footerLanguage": {
+		"string": "Language"
+	},
+	"footerChooseLanguage": {
+		"string": "Choose a language"
+	},
+	"footerScratchLegal": {
+		"string": "Scratch is developed by the Lifelong Kindergarten Group at the MIT Media Lab."
+	},
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply _locales/en_US.json:455]: the string is split in two. the full thing is:
`Scratch is developed by the Lifelong Kindergarten Group at the MIT Media Lab. See http://scratch.mit.edu.`
this is official text published by scratch, so I don't think we should change it

[the remark this answers is not in the document]
Comment (wgyt) [reply _locales/en_US.json:455]: We should tell translators to translate it as shown on Scratch.

[the remark this answers is not in the document]
Comment (RedGuy12) [reply _locales/en_US.json:455]: the scratch tos aren't translated, so we only have an eng ver

[the remark this answers is not in the document]
Comment (wgyt) [reply _locales/en_US.json:455]: oh ok

File: src/auth/discord/index.js (hunk 2 above)
Comment (retronbv) [reply src/auth/discord/index.js:9]: what

**Comment by RedGuy12** [comment]
forgot mobile

**Comment by RedGuy12** [comment]
[this video](https://paul-s-reid.com/web-dev/8f40c9e6-3ddf-4b2f-acb9-07a1923b932d.webm) shows how the footer looks at different scene sizes.
first, it shows how it looks like while scrolling the page
then, it shows what it looks like when hovered.


---

at 767px, the logo disappears and the "copyright 2021" text is moved to the bottom right
at 600px, the bold headers disappear and the language selector goes on its own row
at 475px, lots of links disappear

---

ik about the empty dropdown, it's a bug with translations, will fix after merge (switching languages does work tho!)

**Comment by RedGuy12** [comment]
@wgyt @retronbv pls let me know what you think

will merge after you review

**Comment by GrahamSH-LLK** [comment]
> > * [ ]   move css to tailwind (at)apply
> 
> you just pinged the (at)apply org

Pinging orgs doesn't work that way

**Comment by wgyt** [comment]
> @wgyt @retronbv pls let me know what you think
> 
> will merge after you review

I’m going to review, but iOS can’t play webm files without a third party app

**Comment by wgyt** [comment]
So far, the footer looks great! But why was the video stretched?

**Comment by RedGuy12** [comment]
Not sure, wasn't stretched for me.



________________________________

**Comment by wgyt** [comment]
> Not sure, wasn't stretched for me.

ah, probably a conversion error

**Comment by RedGuy12** [comment]
> > Not sure, wasn't stretched for me.
> 
> ah, probably a conversion error

@wgyt try this one https://youtu.be/sAJDWLHca7E

**Comment by wgyt** [comment]
> > > Not sure, wasn't stretched for me.
> > 
> > 
> > ah, probably a conversion error
> 
> @wgyt try this one https://youtu.be/sAJDWLHca7E

it looks epic, can you finish the todos and then merge it?

**Comment by RedGuy12** [comment]
* [ ]  hide legal info if trademarks are not on page. might not end up being in this pr

not gonna do that rn

**Pull request merged**